torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -152,8 +161,14 @@ def validate_args_and_maybe_create_graph_inputs(
             # Currently, this new input is added to make the calls
             # happy, which expect a fixed number of arguments. In
             # future, we can clean this up.
-            tracer.create_graph_input("const")
-            new_arg = a
+            const_proxy = tracer.create_graph_input("const")
+            if isinstance(a, ConstantVariable) and proxy_for_constants:
+                sym_a = torch.SymInt(a.as_python_constant())
+                const_proxy.node.meta["example_value"] = sym_a
+                sym_arg = SymNodeVariable(const_proxy, sym_a)
+                new_arg = sym_arg
```

Comment:
We should bikeshed on this. I don't have a favorite plan, so I am going to rant off a list of various ways to do this, with pros and cons.

Option 1: This. The benefit is that it opens the door for symbolic args nicely, and we have access to the lowest tracer needed here (current subtracer, aka the tracer input). The con is that its kinda hand-waivey, turning all constants into dynamic values may not be desirable (should always be sound tho). It may even increase the graph break space, meaning less things get captured and more things are likely to fail speculation for no good reason!  I don't really like it.

Option 2: making the sym outside of speculate - where we set store_target for iteration. The benefit is that we are only making a symbolic for what we need, which is nice. The con is that it is *very* annoying to get the subtracer early, and so, it's kind of a PITA to call create_graph_input earlier! We can call it later, but SymNodeVariable should be created with a proxy, and it makes an annoying chicken and egg problem.

Option 3: User defined mapping of which constants to turn into symints. No real downside, but kind of an annoying API to use, I guess?

Option 4: RangeIteratorVariable always holds a symbolic value, with a proxy that is a node outside of the speculated subgraph.

Option 5: Alt alt approach is we don't use symbolic shapes, but produce a new light-weight proxyable int, where we can shuffle the proxy around and have it balance being an input to the subgraph without being an input to the parent (must be a named node in the parent tho, a la maybe_fx_node() below)

Thoughts @rzou?